packages/mds-api-authorizer/index.ts

```diff
@@ -14,11 +14,41 @@ const {
   TOKEN_USER_EMAIL_CLAIM = 'https://ladot.io/user_email'
 } = process.env
 
-export type ApiAuthorizer = (req: express.Request) => ApiAuthorizerClaims | null
+export type ApiAuthorizer = (req: express.Request | string) => ApiAuthorizerClaims | null
 
 export const AuthorizationHeaderApiAuthorizer: ApiAuthorizer = req => {
-  if (req.headers && req.headers.authorization) {
-    const [scheme, token] = req.headers.authorization.split(' ')
+  if (typeof req !== 'string') {
+    if (req.headers && req.headers.authorization) {
+      const [scheme, token] = req.headers.authorization.split(' ')
+      const decoders: { [scheme: string]: () => ApiAuthorizerClaims } = {
+        bearer: () => {
+          const {
+            sub: principalId,
+            scope,
+            [TOKEN_PROVIDER_ID_CLAIM]: provider_id = null,
+            [TOKEN_USER_EMAIL_CLAIM]: user_email = null,
+            ...claims
+          } = decode(token)
+          return { principalId, scope, provider_id, user_email, ...claims }
+        },
+        basic: () => {
+          const [principalId, scope] = Buffer.from(token, 'base64')
+            .toString()
+            .split('|')
+          return { principalId, scope, provider_id: principalId, user_email: null }
+        }
+      }
+      const decoder = decoders[scheme.toLowerCase()]
+      return decoder ? decoder() : null
```

Comment:
Could we rename some of these variables to make the code more readable?

Reply:
I think my refactor helps address this a bit. We gucci?